go.mod

```diff
@@ -1,9 +1,9 @@
 module github.com/oauth2-proxy/oauth2-proxy/v7
 
-go 1.22.0
+go 1.22.7
```

Comment:
Do we need to bump this?

Reply:
Yes and no. 

If we leave it at 1.22.0, it will still use the latest patch version during CI builds but when manually building it with the Makefile and go1.22.7 installed. It will modify the `go.mod` file and add another line `toolchain go1.22.7` below it.

Reply:
https://go.dev/doc/toolchain

Reply:
With the introduction of toolchain handling in go1.21 they already addressed this:

It automatically downloads the correct golang version like any other pkg:

![image](https://github.com/user-attachments/assets/ebb417f1-b027-470a-a451-c64962921a87)

![image](https://github.com/user-attachments/assets/c571cf20-a0e4-448c-b0d5-cccecc377033)

> The go command can use its bundled Go toolchain as well as other versions that it finds in the local PATH or downloads as needed.
The choice of Go toolchain being used depends on the GOTOOLCHAIN environment setting and the go and toolchain lines in the main module’s go.mod file or the current workspace’s go.work file. As you move between different main modules and workspaces, the toolchain version being used can vary, just as module dependency versions do.
https://go.dev/doc/toolchain


![image](https://github.com/user-attachments/assets/9c4f1b54-8935-4004-9671-4a4e2ca46432)